google/cloud/spanner/integration_tests/backup_integration_test.cc

```diff
@@ -36,116 +37,115 @@ namespace spanner {
  protected:
-  void SetUp() override {
-    BackupTest::SetUp();
-    instance_name_regex_ = std::regex(
-        R"(projects/.+/instances/(temporary-instance-(\d{4}-\d{2}-\d{2})-.+))");
-    instance_config_regex_ = std::regex(".*us-west.*");
-    if (!run_slow_backup_tests_) {
-      return;
-    }
-    auto s = spanner_testing::CleanupStaleInstances(project_id_,
-                                                    instance_name_regex_);
-    EXPECT_STATUS_OK(s) << s.message();
-  }
-  std::regex instance_name_regex_;
-  std::regex instance_config_regex_;
+  google::cloud::internal::DefaultPRNG generator_;
+  DatabaseAdminClient database_admin_client_;
 };
 
 /// @test Backup related integration tests.
-TEST_F(BackupTestWithCleanup, BackupTestSuite) {
-  if (!run_slow_backup_tests_ || emulator_) {
-    GTEST_SKIP();
-  }
-  auto generator = google::cloud::internal::MakeDefaultPRNG();
-  std::string instance_id =
-      google::cloud::spanner_testing::RandomInstanceName(generator);
-  std::string database_id =
-      google::cloud::spanner_testing::RandomDatabaseName(generator);
-  std::string restore_database_id =
-      google::cloud::spanner_testing::RandomDatabaseName(generator);
-  ASSERT_FALSE(project_id_.empty());
-  ASSERT_FALSE(instance_id.empty());
-  Instance in(project_id_, instance_id);
-
-  // Make sure we're using correct regex.
-  std::smatch m;
-  auto full_name = in.FullName();
-  EXPECT_TRUE(std::regex_match(full_name, m, instance_name_regex_));
-  std::string instance_config = spanner_testing::PickInstanceConfig(
-      project_id_, instance_config_regex_, generator);
-  ASSERT_FALSE(instance_config.empty()) << "could not get an instance config";
-  future<StatusOr<google::spanner::admin::instance::v1::Instance>> f =
-      instance_admin_client_.CreateInstance(
-          CreateInstanceRequestBuilder(in, instance_config)
-              .SetDisplayName("test-display-name")
-              .SetNodeCount(1)
-              .Build());
-  auto instance = f.get();
+TEST_F(BackupTest, BackupTest) {
+  if (!RunSlowBackupTests() || Emulator()) GTEST_SKIP();
 
-  EXPECT_STATUS_OK(instance);
+  InstanceAdminClient instance_admin_client(MakeInstanceAdminConnection(
+      ConnectionOptions{}, spanner_testing::TestRetryPolicy(),
+      spanner_testing::TestBackoffPolicy(),
+      spanner_testing::TestPollingPolicy()));
 
-  Database db(project_id_, instance_id, database_id);
-
-  auto database_future = database_admin_client_.CreateDatabase(db);
-  auto database = database_future.get();
+  Instance in(ProjectId(), spanner_testing::RandomInstanceName(generator_));
+  auto instance_config = spanner_testing::PickInstanceConfig(
+      in.project_id(), std::regex(".*us-west.*"), generator_);
+  ASSERT_FALSE(instance_config.empty()) << "could not get an instance config";
+  EXPECT_STATUS_OK(
+      instance_admin_client
+          .CreateInstance(CreateInstanceRequestBuilder(in, instance_config)
```

Comment:
Do you know why this test needs a new instance?  I mean, it is going to create databases and backups with random names, an existing instance should be able to do that?  Or is there some limitation on the number of backups per instance we need to workaround?

Reply:
No, I don't know.  My thought is there was no better way to ensure a "us-west" instance (for whatever reason that was).

I am working on a filtering capability for `PickRandomInstance()`, which will be required for the CMEK merge (at least for the tests in their current state).  See #5588.  At that time I can also simplify this (if we believe "us-west" is still a feature).